src/messages/mod.rs

```diff
@@ -78,19 +78,68 @@ impl HopMessage {
     }
 }
 
+/// Metadata needed for verification of the sender
+#[derive(Ord, PartialOrd, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
+pub struct PartialSecurityMetadata {
+    proof: SectionProofChain,
+    sender_prefix: Prefix<XorName>,
+    shares: BTreeMap<BlsPublicKeyShare, BlsSignatureShare>,
+    pk_set: BlsPublicKeySet,
+}
+
+/// Metadata needed for verification of the sender
+#[derive(Ord, PartialOrd, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
+pub struct FullSecurityMetadata {
+    proof: SectionProofChain,
+    sender_prefix: Prefix<XorName>,
+    signature: BlsSignature,
+}
+
+impl FullSecurityMetadata {
+    pub fn verify_sig(&self, bytes: &[u8]) -> bool {
+        self.proof.last_public_key().verify(&self.signature, bytes)
+    }
+
+    pub fn validate_proof(&self) -> bool {
+        self.proof.validate()
+    }
+
+    pub fn prefix(&self) -> &Prefix<XorName> {
+        &self.sender_prefix
+    }
+
+    pub fn proof_chain(&self) -> &SectionProofChain {
+        &self.proof
+    }
+}
+
+impl Debug for FullSecurityMetadata {
+    fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
+        write!(
+            formatter,
+            "FullSecurityMetadata {{ sender_prefix: {:?}, proof: {:?}, .. }}",
+            self.sender_prefix, self.proof
+        )
+    }
+}
+
+#[derive(Ord, PartialOrd, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
```

Comment:
better have a comment explains why there is partial and full security metadata, and how they shall be used differently?
I see some fields of them are duplicated?

Reply:
Will update